### PR TITLE
Video upload complete lambda

### PR DESCRIPTION
**Pull request by angellee-io**

### CODE REVIEW

File: src/com/grideo/videoProcessing/requestHandlers/VideoUploadCompletedRequestHandler.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	public VideoUploadCompletedRequestHandler() {
+		databaseManager = DatabaseManager.getInstance();
+	}
+	
+	public VideoUploadCompletedRequestHandler(DatabaseManager manager) {
+		this.databaseManager = manager;
+	}
+	
+	@Override
+	public Response handleRequest(S3Event input, Context context) {		LOG.info("VideoUploadCompletedRequestHandler");
```
Comment (aaljebouri) [inline src/com/grideo/videoProcessing/requestHandlers/VideoUploadCompletedRequestHandler.java]: This Log.info is in a fun place?

File: src/com/grideo/videoProcessing/invoker/LambdaInvoker.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.lambda.AWSLambdaAsync;
+import com.amazonaws.services.lambda.AWSLambdaAsyncClientBuilder;
+import com.amazonaws.services.lambda.model.InvocationType;
+import com.amazonaws.services.lambda.model.InvokeRequest;
+
+public class LambdaInvoker {
+
+    public void runWithoutPayload(String region, String functionName) {
```
Comment (aaljebouri) [inline src/com/grideo/videoProcessing/invoker/LambdaInvoker.java]: I don't see anywhere where this function is used.

File: src/com/grideo/videoProcessing/requestHandlers/VideoUploadCompletedRequestHandler.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+	}
+	
+	@Override
+	public Response handleRequest(S3Event input, Context context) {		LOG.info("VideoUploadCompletedRequestHandler");		
+		throwExceptionIfInputIsNotValid(input);
+		
+		record = input.getRecords().get(0);		
+		throwExceptionIfRecordIsNotValid();
+
+        String s3Key = record.getS3().getObject().getKey();
```
Comment (aaljebouri) [inline src/com/grideo/videoProcessing/requestHandlers/VideoUploadCompletedRequestHandler.java]: Can this be it's own function? determine fileIdBasedOnS3UploadKey?

File: src/com/grideo/videoProcessing/requestHandlers/VideoUploadCompletedRequestHandler.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        
+        GetVideoByFileIdQuery getQuery = new GetVideoByFileIdQuery(fileId);
+        Video videoInDatabase = databaseManager.execute(getQuery); 
+        videoInDatabase.setWasUploadCompleted(true);		
+		videoInDatabase.setUploadedOn(new Date());
+
+		Integer currentPosts = videoInDatabase.getOwner().getTotalPosts();
+		videoInDatabase.getOwner().setTotalPosts(++currentPosts);
+
+		UpdateVideoAndOwnerQuery query = new UpdateVideoAndOwnerQuery(
```
Comment (aaljebouri) [inline src/com/grideo/videoProcessing/requestHandlers/VideoUploadCompletedRequestHandler.java]: Can from GetVideoByFileId to UpdateVideoAndOwnerQuery go into it's own function? probably in a separate class.

File: src/com/grideo/videoProcessing/requestHandlers/VideoUploadCompletedRequestHandler.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+		Boolean uploaded = databaseManager.execute(query);
+		
+		if (uploaded) {
+			invokeEncoderLambda(input);
+		}
+				
+		return new Response(uploaded);
+	}
+	
+	private void invokeEncoderLambda(S3Event input) {
```
Comment (aaljebouri) [inline src/com/grideo/videoProcessing/requestHandlers/VideoUploadCompletedRequestHandler.java]: Can this logic be extracted too?

File: src/com/grideo/videoProcessing/requestHandlers/VideoUploadCompletedRequestHandler.java (hunk 4 above)
Comment (angellee-io) [reply src/com/grideo/videoProcessing/requestHandlers/VideoUploadCompletedRequestHandler.java]: I feel like it's a bit too much to separate them to a class.. i will put them in a method.

**Pull request merged**